### ISSUE TITLE
The --unlock option may not be working

### ISSUE THREAD

**Issue by mikeseese** [ConsenSys-archive/ganache-cli-archive#535]
Per the comment from @elenadimitrova (https://github.com/trufflesuite/ganache-cli/issues/526#issuecomment-395339685), it appeared there are issues with using the `--unlock` option (at a minimum, by itself).

The version used was `6.1.0` for the below output:
```
/Users/Elena/colonyNetwork/node_modules/ganache-cli/build/cli.node.js:2
!function(e){var t={};function a(n){if(t[n])return t[n].exports;var r=t[n]={i:n,l:!1,exports:{}};return e[n].call(r.exports,r,r.exports,a),r.l=!0,r.exports}a.m=e,a.c=t,a.d=function(e,t,n){a.o(e,t)||Object.defineProperty(e,t,{configurable:!1,enumerable:!0,get:n})},a.r=function(e){Object.defineProperty(e,"__esModule",{value:!0})},a.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return a.d(t,"a",t),t},a.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},a.p="",a.w={},a(a.s=1301)}([function(e,t,a){try{var n=a(1);if("function"!=typeof n.inherits)throw"";e.exports=n.inherits}catch(t){e.exports=a(1218)}},function(e,t){e.exports=require("util")},function(e,t){e.exports=require("path")},function(e,t,a){"use strict";var n="function"==typeof Symbol&&"symbol"==typeof Symbol.iterator?function(e){return typeof e}:function(e){return e&&"function"==typeof Symbol&&e.constructor===Symbol&&e!==Symbol.prototype?"symbol":typeof e},r=a(1221),c=a(110),i=a(13),f=a(65),o=a(12),d=a(34),s=a(10).Buffer;Object.assign(t,a(438)),t.MAX_INTEGER=new o("ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",16),t.TWO_POW256=new o("10000000000000000000000000000000000000000000000000000000000000000",16),t.SHA3_NULL_S="c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470",t.SHA3_NULL=s.from(t.SHA3_NULL_S,"hex"),t.SHA3_RLP_ARRAY_S="1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",t.SHA3_RLP_ARRAY=s.from(t.SHA3_RLP_ARRAY_S,"hex"),t.SHA3_RLP_S="56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",t.SHA3_RLP=s.from(t.SHA3_RLP_S,"hex"),t.BN=o,t.rlp=f,t.secp256k1=c,t.zeros=function(e){return s.allocUnsafe(e).fill(0)},t.zeroAddress=function(){var e=t.zeros(20);return t.bufferToHex(e)},t.setLengthLeft=t.setLength=function(e,a,n){var r=t.zeros(a);return e=t.toBuffer(e),n?e.length<a?(e.copy(r),r):e.slice(0,a):e.length<a?(e.copy(r,a-e.length),r):e.slice(-a)},t.setLengthRight=function(e,a){return t.setLength(e,a,!0)},t.unpad=t.stripZeros=function(e){for(var a=(e=t.stripHexPrefix(e))[0];e.length>0&&"0"===a.toString();)a=(e=e.slice(1))[0];return e},t.toBuffer=function(e){if(!s.isBuffer(e))if(Array.isArray(e))e=s.from(e);else if("string"==typeof e)e=t.isHexString(e)?s.from(t.padToEven(t.stripHexPrefix(e)),"hex"):s.from(e);else if("number"==typeof e)e=t.intToBuffer(e);else if(null===e||void 0===e)e=s.allocUnsafe(0);else if(o.isBN(e))e=e.toArrayLike(s);else{if(!e.toArray)throw new Error("invalid type");e=s.from(e.toArray())}return e},t.bufferToInt=function(e){return new o(t.toBuffer(e)).toNumber()},t.bufferToHex=function(e){return"0x"+(e=t.toBuffer(e)).toString("hex")},t.fromSigned=function(e){return new o(e).fromTwos(256)},t.toUnsigned=function(e){return s.from(e.toTwos(256).toArray())},t.sha3=function(e,a){return e=t.toBuffer(e),a||(a=256),r("keccak"+a).update(e).digest()},t.sha256=function(e){return e=t.toBuffer(e),d("sha256").update(e).digest()},t.ripemd160=function(e,a){e=t.toBuffer(e);var n=d("rmd160").update(e).digest();return!0===a?t.setLength(n,32):n},t.rlphash=function(e){return t.sha3(f.encode(e))},t.isValidPrivate=function(e){return c.privateKeyVerify(e)},t.isValidPublic=function(e,t){return 64===e.length?c.publicKeyVerify(s.concat([s.from([4]),e])):!!t&&c.publicKeyVerify(e)},t.pubToAddress=t.publicToAddress=function(e,a){return e=t.toBuffer(e),a&&64!==e.length&&(e=c.publicKeyConvert(e,!1).slice(1)),i(64===e.length),t.sha3(e).slice(-20)};var b=t.privateToPublic=function(e){return e=t.toBuffer(e),c.publicKeyCreate(e,!1).slice(1)};t.importPublic=function(e){return 64!==(e=t.toBuffer(e)).length&&(e=c.publicKeyConvert(e,!1).slice(1)),e},t.ecsign=function(e,t){var a=c.sign(e,t),n={};return n.r=a.signature.slice(0,32),n.s=a.signature.slice(32,64),n.v=a.recovery+27,n},t.hashPersona
```

I then testing on `6.1.3` and also experienced erroneous behavior. We will need to investigate this issue further to see what/if there is a problem

**Comment by andy-mcewan-artos-io** [comment]
Yep, unlock isn't working for me with 6.1.4. Have rolled back to 6.1.0 til this is fixed.

**Comment by benjamincburns** [comment]
@elenadimitrova's issue appears to have been caused by using the `--unlock` flag with no argument. We will add error checking to that. Otherwise I've tested the --unlock flag with a proper account argument on the latest release and it's working fine - just a really ugly error message when you don't specify the account to unlock

**Comment by mikeseese** [comment]
Looks like there was a bug with the unlock option; it'll be in the next release

**Comment by tpikachu** [comment]
Is there any update on this issue?

**Comment by mikeseese** [comment]
@tpikachu, it looks like I fixed this in #551 which was released in `ganache-cli@6.8.3`, so this bug should be fixed in the most recent version. If you're still having issues, please open a new issue. Thanks!
